modules/hosted-cluster-etcd-quorum-loss-recovery.adoc

```diff
@@ -0,0 +1,260 @@
+
+If multiple members of the etcd cluster have lost data or return a `CrashLoopBackOff` status, it can cause an etcd quorum loss. You must restore your etcd cluster from a snapshot.
+
+[IMPORTANT]
+====
+This procedure requires API downtime.
+====
+
+.Prerequisites
+* The `oc` and `jq` binaries have been installed.
+
+.Procedure
+
+. Set up environment variables for your hosted cluster by entering the following commands, replacing values as necessary:
++
+[source,terminal]
+----
+$ CLUSTER_NAME=my-cluster
+$ HOSTED_CLUSTER_NAMESPACE=clusters
+$ CONTROL_PLANE_NAMESPACE="${HOSTED_CLUSTER_NAMESPACE}-${CLUSTER_NAME}"
+----
+
+. Pause reconciliation of the hosted cluster by entering the following command, replacing values as necessary:
++
+[source,terminal]
+----
+$ oc patch -n ${HOSTED_CLUSTER_NAMESPACE} hostedclusters/${CLUSTER_NAME} -p '{"spec":{"pausedUntil":"true"}}' --type=merge
+----
+
+. Scale down API servers by entering the following commands:
++
+[source,terminal]
+----
+$ oc scale -n ${CONTROL_PLANE_NAMESPACE} deployment/kube-apiserver --replicas=0
+$ oc scale -n ${CONTROL_PLANE_NAMESPACE} deployment/openshift-apiserver --replicas=0
+$ oc scale -n ${CONTROL_PLANE_NAMESPACE} deployment/openshift-oauth-apiserver --replicas=0
+----
+
+. Take a snapshot of etcd by using one of the following methods:
+
+.. Use a previously backed up snapshot of etcd
+.. If you have an available etcd pod, take a snapshot from the active etcd pod by completing the following steps:
++
+... List etcd pods by entering the following command:
++
+[source,terminal]
+----
+`$ oc get -n ${CONTROL_PLANE_NAMESPACE} pods -l app=etcd`
+----
++
+... Take a snapshot of the pod database and save it locally to your machine by entering the following command:
++
+[source,terminal]
++
+----
+ETCD_POD=etcd-0 
+oc exec -n ${CONTROL_PLANE_NAMESPACE} -c etcd -t ${ETCD_POD} -- env ETCDCTL_API=3 /usr/bin/etcdctl \
+--cacert /etc/etcd/tls/etcd-ca/ca.crt \
+--cert /etc/etcd/tls/client/etcd-client.crt \
+--key /etc/etcd/tls/client/etcd-client.key \
+--endpoints=https://localhost:2379 \
+snapshot save /var/lib/snapshot.db
+----
++
+... Verify that the snapshot is successful by entering the following command:
++
+[source,terminal]
+----
+`oc exec -n ${CONTROL_PLANE_NAMESPACE} -c etcd -t ${ETCD_POD} -- env ETCDCTL_API=3 /usr/bin/etcdctl -w table snapshot status /var/lib/snapshot.db`
+----
++
+... Make a local copy of the snapshot by entering the following command:
++
+[source,terminal]
+----
+`oc cp -c etcd ${CONTROL_PLANE_NAMESPACE}/${ETCD_POD}:/var/lib/snapshot.db /tmp/etcd.snapshot.db`
+----
++
+... Make a copy of the snapshot database from etcd persistent storage.
++
+... List etcd pods by entering the following command:
++
+[source,terminal]
+----
+`oc get -n ${CONTROL_PLANE_NAMESPACE} pods -l app=etcd` 
+----
++
+... Find a pod that is running and set its name as the value of `ETCD_POD ETCD_POD=etcd-0`, then copy its snapshot database by entering the following command:
```

Comment:
change: set its name as the value of `ETCD_POD ETCD_POD=etcd-0`
to: set its name as the value of `ETCD_POD`:  `ETCD_POD=etcd-0`